lib/Backends/CPU/CPUDeviceManager.cpp

```diff
@@ -113,10 +113,11 @@ void CPUDeviceManager::evictNetworkImpl(std::string functionName,
 void CPUDeviceManager::runFunctionImpl(
     RunIdentifierTy id, std::string function,
     std::unique_ptr<ExecutionContext> context, ResultCBTy resultCB) {
-  context->logTraceEvent("DM_run", "B");
+  std::string eventName = "run_" + function + "_" + std::to_string(id);
+  TRACE_EVENT_BEGIN(context->getTraceContext(), eventName);
   auto funcIt = functions_.find(function);
   if (funcIt == functions_.end()) {
-    context->logTraceEvent("DM_run", "E", {{"reason", "function not found"}});
+    context->logTraceEvent(eventName, "E", {{"reason", "function not found"}});
```

Comment:
Consider making a `TRACE_EVENT_{BEGIN/END}_LOG` macro as well. I know what `"E"` represents because the definition of `TRACE_EVENT_END` is right above it in the list of changes but I don't think it would be as obvious if I looked at this file only.

Consider also a set of `constexpr` strings in the `TraceEvent` namespace for `"B", "E", and "I"` that you can reuse in the `TRACE_EVENT_XXX` macros. Something like this:

```
struct TraceEvent {
...
static constexpr auto BeginType = "B";
static constexpr auto InstantType = "I";
static constexpr auto EndType = "E";
...
};

...

#define TRACE_EVENT_BEGIN(ctx, name)                                           \
  if (ctx) {                                                                   \
    ctx->logTraceEvent(name, TraceEvent::BeginType);                           \
  }

...
```

Then you could write this as `context->logTraceEvent(eventName, TraceEvent::EndType, /*args=*/{{"reason", "function not found"}});`.